CMakeLists.txt

```diff
@@ -159,12 +159,12 @@ endif()
 
```

Comment:
With this refactoring, we *should* be able to remove `NOVELRT_HEADERS` and move it to be exclusively in `NOVELRT_LIBRARY_HEADERS`.

`NOVELRT_EXECUTABLE_HEADERS` would then be mostly empty and would find the files via `include_directories`, which makes it match what external users would do.

Reply:
(we can do this in a separate PR if desired)